horovod/torch/functions.py

```diff
@@ -68,6 +67,7 @@ def broadcast_optimizer_state(optimizer, root_rank):
         root_rank: The rank of the process from which the optimizer will be
                    broadcasted to all other processes.
     """
+    from horovod.torch.optimizer import DistributedOptimizer
```

Comment:
Why is this here?

Reply:
I had issues when I tried importing `broadcast_object` in `optimizer.py` due to a circular dependency on `DistributedOptimizer`. To fix this, I limited the import of `DistributedOptimizer` in this file to the scope of the function that needs it only.

Reply:
Ah, gotcha.  We can probably move things around in a follow-up PR to address that, but this is good for now.